app/controllers/users_controller.rb

```diff
@@ -0,0 +1,29 @@
+class UsersController < ApplicationController
+    def show
+        @user = User.find_by id: params[:id]
+        return if @user
+
+        redirect_to root_path

```

Comment:
chỗ này trước khi redirect thì thêm flash message lỗi không tồn tại user nữa nhé
```suggestion
        flash[:error] = "Error ...."
        redirect_to root_path
```